web/src/main/java/com/navercorp/pinpoint/web/controller/MapController.java

```diff
@@ -256,54 +262,30 @@ public NodeHistogramSummary postResponseTimeHistogramDataV2(
             @RequestParam("serviceTypeCode") Short serviceTypeCode,
             @RequestParam("from") long from,
             @RequestParam("to") long to,
-            @RequestBody ApplicationPair applicationPair) {
+            @RequestBody ApplicationPairs applicationPairs) {
         final Range range = new Range(from, to);
         dateLimit.limit(range);
 
         Application application = applicationFactory.createApplication(applicationName, serviceTypeCode);
 
-        Application fromApplication = null;
-        String fromApplicationName = applicationPair.getFromApplicationName();
-        if (!StringUtils.isEmpty(fromApplicationName)) {
-            fromApplication = applicationFactory.createApplication(fromApplicationName, applicationPair.getFromServiceTypeCode());
-        }
-
-        Application toApplication = null;
-        String toApplicationName = applicationPair.getToApplicationName();
-        if (!StringUtils.isEmpty(toApplicationName)) {
-            toApplication = applicationFactory.createApplication(toApplicationName, applicationPair.getToServiceTypeCode());
-        }
+        List<Application> fromApplications = mapApplicationPairsToApplications(applicationPairs.getFromApplications());
+        List<Application> toApplications = mapApplicationPairsToApplications(applicationPairs.getToApplications());
 
-        return responseTimeHistogramService.selectNodeHistogramData(application, range, fromApplication, toApplication);
+        return responseTimeHistogramService.selectNodeHistogramData(application, range, fromApplications, toApplications);
     }
 
-    @RequestMapping(value = "/getResponseTimeHistogramDataV2", method = RequestMethod.GET)
```

Comment:
The plan was to use POST only as from / to application parameters may be too long for GETs.
But still having a GET API available would be a good idea,
Will add it back, thanks!